### DISCUSSION TITLE
applying-the-routedatarequest-cultureprovider-globally-with-middleware-as-filters/

### DISCUSSION THREAD

**Discussion by andrewlock** [andrewlock/blog-comments#181] [Announcements]
# applying-the-routedatarequest-cultureprovider-globally-with-middleware-as-filters/

In this post, I'll show the various ways you can configure your routes globally, so that all your urls will have a culture prefix by default.

https://andrewlock.net/applying-the-routedatarequest-cultureprovider-globally-with-middleware-as-filters/

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at December 20 2016, 01:32, in reply to [Harald Ilisch](https://disqus.com/by/haraldilisch/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/181#discussioncomment-6024645)</em>

---

<p>Hi Harald, thanks for reading and the comment:) That article is great, as it deals with some of the issues in combining attribute routes with global routes. Unfortunately, I don't think it deals with the breaking of global conventions with the IApplicationModelConvention above, as that essentially adds a route attribute to every controller.</p><p>Controllers that were previously accessible thanks to the global convention will no longer be available, as they have an attribute route! I'm still looking for a good solution, but if you find a way around it, do let me know. Thanks!</p>

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at January 13 2018, 06:18, in reply to [Andrew Boyd](https://disqus.com/by/disqus_xRnCUcO1iz/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/181#discussioncomment-6024650)</em>

---

<p>I think you probably want to take a look at the next post in the series :) <a href="https://andrewlock.net/using-a-culture-constraint-and-catching-404s-with-the-url-culture-provider/" rel="nofollow noopener" title="https://andrewlock.net/using-a-culture-constraint-and-catching-404s-with-the-url-culture-provider/">https://andrewlock.net/using-a-culture-constraint-and-catching-404s-with-the-url-culture-provider/</a></p>

[a comment by an automated account is not shown]

**Comment by TuralHasanov117** [comment]
Hello Andrew,

If we are using api endpoints, does it make sense to store "culture" preference in request headers instead of api url? 
I think setting Accept-Language header to the culture we want in request header would be better approach in terms of security and api routes will not be dependent on any culture segment.
